File: vbnet/list_merger/list_merger_6.vb
' This program is called List Merger.
' The purpose of this program is to merge two lists of integers into a single list.
' The program will prompt the user to enter two lists of integers, separated by commas.
' It will then merge these lists and display the merged list.
' The program is designed to help programmers understand how to handle user input and merge lists.

Module ListMerger

    Sub Main()
        ' Declare variables to hold the user input and the merged list
        Dim input1 As String
        Dim input2 As String
        Dim list1 As List(Of Integer)
        Dim list2 As List(Of Integer)
        Dim mergedList As List(Of Integer)

        ' Prompt the user to enter the first list of integers
        Console.WriteLine("Enter the first list of integers, separated by commas:")
        input1 = Console.ReadLine()

        ' Convert the first input string to a list of integers
        list1 = ConvertInputToList(input1)

        ' Prompt the user to enter the second list of integers
        Console.WriteLine("Enter the second list of integers, separated by commas:")
        input2 = Console.ReadLine()

        ' Convert the second input string to a list of integers
        list2 = ConvertInputToList(input2)

        ' Merge the two lists
        mergedList = MergeLists(list1, list2)

        ' Display the merged list
        Console.WriteLine("The merged list is:")
        For Each num As Integer In mergedList
            Console.Write(num & " ")
        Next
        Console.WriteLine()

    End Sub

    ' Function to convert a comma-separated string to a list of integers
    Function ConvertInputToList(ByVal input As String) As List(Of Integer)
        Dim result As New List(Of Integer)
        Dim elements() As String = input.Split(","c)

        For Each element As String In elements
            result.Add(Convert.ToInt32(element.Trim()))
        Next

        Return result
    End Function

    ' Function to merge two lists of integers
    Function MergeLists(ByVal list1 As List(Of Integer), ByVal list2 As List(Of Integer)) As List(Of Integer)
        Dim result As New List(Of Integer)
        result.AddRange(list1)
        result.AddRange(list2)
        Return result
    End Function

End Module

File: rust/list_merger/list_merger_6.rs
// This program is called list merger. It is designed to merge two lists of integers into a single list.
// The program will take two lists as input from the user, merge them, and then print the merged list.
// The purpose of this program is to help programmers practice their Rust skills and understand how to handle lists and user input.

use std::io;

fn main() {
    // Prompt the user for the first list of integers
    println!("Enter the first list of integers, separated by spaces:");
    let mut input1 = String::new();
    io::stdin().read_line(&mut input1).expect("Failed to read line");

    // Parse the input into a vector of integers
    let list1: Vec<i32> = input1
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Parse error"))
        .collect();

    // Prompt the user for the second list of integers
    println!("Enter the second list of integers, separated by spaces:");
    let mut input2 = String::new();
    io::stdin().read_line(&mut input2).expect("Failed to read line");

    // Parse the input into a vector of integers
    let list2: Vec<i32> = input2
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Parse error"))
        .collect();

    // Merge the two lists
    let mut merged_list = merge_lists(list1, list2);

    // Print the merged list
    println!("The merged list is: {:?}", merged_list);
}

// Function to merge two lists of integers
fn merge_lists(list1: Vec<i32>, list2: Vec<i32>) -> Vec<i32> {
    let mut result = Vec::new();
    for &item in list1.iter() {
        result.push(item);
    }
    for &item in list2.iter() {
        result.push(item);
    }
    result
}

// Function to calculate the sum of a list of integers
fn sum_list(list: Vec<i32>) -> i32 {
    let mut sum = 0;
    for &item in list.iter() {
        sum += item;
    }
    sum
}

// Function to find the maximum value in a list of integers
fn max_list(list: Vec<i32>) -> i32 {
    let mut max = list[0];
    for &item in list.iter() {
        if item > max {
            max = item;
        }
    }
    max
}

// Function to find the minimum value in a list of integers
fn min_list(list: Vec<i32>) -> i32 {
    let mut min = list[0];
    for &item in list.iter() {
        if item < min {
            min = item;
        }
    }
    min
}

// Function to calculate the average of a list of integers
fn average_list(list: Vec<i32>) -> f32 {
    let sum = sum_list(list.clone());
    let count = list.len();
    sum as f32 / count as f32
}

// Function to sort a list of integers in ascending order
fn sort_list(mut list: Vec<i32>) -> Vec<i32> {
    list.sort();
    list
}

// Function to reverse a list of integers
fn reverse_list(mut list: Vec<i32>) -> Vec<i32> {
    list.reverse();
    list
}

// Function to check if a list of integers is sorted in ascending order
fn is_sorted(list: Vec<i32>) -> bool {
    for i in 0..list.len() - 1 {
        if list[i] > list[i + 1] {
            return false;
        }
    }
    true
}

// Function to remove duplicates from a list of integers
fn remove_duplicates(list: Vec<i32>) -> Vec<i32> {
    let mut result = Vec::new();
    for &item in list.iter() {
        if !result.contains(&item) {
            result.push(item);
        }
    }
    result
}

// Function to find the median of a list of integers
fn median_list(mut list: Vec<i32>) -> f32 {
    list.sort();
    let len = list.len();
    if len % 2 == 0 {
        (list[len / 2 - 1] + list[len / 2]) as f32 / 2.0
    } else {
        list[len
         / 2] as f32
    }
}

// Function to find the mode of a list of integers
fn mode_list(list: Vec<i32>) -> i32 {
    let mut occurrences = std::collections::HashMap::new();
    for &item in list.iter() {
        *occurrences.entry(item).or_insert(0) += 1;
    }
    *occurrences.iter().max_by_key(|&(_, count)| count).map(|(&key, _)| key).unwrap()
}

// Function to find the range of a list of integers